arrow-ipc/src/reader.rs

```diff
@@ -79,6 +83,16 @@ fn create_array(reader: &mut ArrayReader, field: &Field) -> Result<ArrayRef, Arr
                 reader.next_buffer()?,
             ],
         ),
+        BinaryView | Utf8View => {
+            let count = variadic_counts
+                .pop_front()
+                .ok_or(ArrowError::IpcError("Incorrect variadic count!".to_owned()))?;
```

Comment:
I think being more explicit would help here. Maybe something like

```suggestion
                .ok_or(ArrowError::IpcError(format!("{data_type} require at least one variadic count, got none"))?;
```

Reply:
Even if we have some variadic counts we could hit this error if there were more columns requiring them.
```suggestion
                .ok_or(ArrowError::IpcError(format!("Missing variadic count for {data_type} column")))?;
```